hw/ip/otp_ctrl/data/otp_ctrl.hjson.tpl

```diff
@@ -560,6 +560,67 @@
     }
   ]
 
+  features: [
+    {
+      name: "OTP_CTRL.VENDOR_TEST_PARTITION"
+      desc: '''Vendor test partition is intended to be used for OTP programming smoke check during manufacturing flow.
+      In this partition, uncorrectable errors will not lead to fatal errors and alerts.
+      '''
+    }
+    {
+      name: "OTP_CTRL.CALIBRATION_TEST"
+      desc: '''During calibration stage, various parameters (clock, voltage, and timing sources) are calibrated and recorded to CREATOR_SW_CFG partition.
+      '''
+    }
+    {
+      name: "OTP_CTRL.INIT"
+      desc: '''When power is up, OTP controller reads devices status.
+      After all reads complete, the controller performs integrity check on the HW_CFG and SECRET partitions.
+      Once all integrity checks are complete, the controller marks outputs as valid.
+      '''
+    }
+    {
+      name: "OTP_CTRL.ENTROPY_READ"
+      desc: '''Firmware can read entropy from ENTROPY_SRC block by configuring following field of HW_CFG partition.
+        - EN_CSRNG_SW_APP_READ
+        - EN_ENTROPY_SRC_FW_READ
+        - EN_ENTROPY_SRC_FW_OVER
+      '''
+    }
+    {
+      name: "OTP_CTRL.KEY_DERIVATION"
+      desc: "OTP controller participate key derivation process by providing scramble key seed to SRAM_CTRL and FLASH_CTRL."
+    }
+    {
+      name: "OTP_CTRL.PROGRAM"
+      desc: '''All other partitions except life cycle partition are programmed through DAI interface.
+      And once non-zero digest is programmed to these partition, no further write access is allowed.
+      Life cycle partition is programmed by LC_CTRL.
+      '''
+    }
+    {
```

Comment:
I wonder if it is better to group the secret partitions in a FUNCTIONAL_GROUP, and have sub-features for read_lock (readable before digest update, not readable after). I suppose we test dai read access before digest update.